### DISCUSSION TITLE
Working sample w. DataTriggerBehavior

### DISCUSSION THREAD

**Discussion by elveejay1** [AvaloniaUI/Avalonia#8636] [Q&A] [answered]
I've tried to experiment a bit with DataTriggerBehavior, but VS 2022 keeps telling me:
Unable to resolve type DataTriggerBehvior from namespace clr-namespace:Avalonia.Xaml.Interactions.Core;assembly=Avalonia.Xaml.Interactions (line 30 position 6) Line 30, position 6.	
My xaml:

`
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:i="clr-namespace:Avalonia.Xaml.Interactivity;assembly=Avalonia.Xaml.Interactivity"
             xmlns:ia="clr-namespace:Avalonia.Xaml.Interactions.Core;assembly=Avalonia.Xaml.Interactions"
             xmlns:iac="clr-namespace:Avalonia.Xaml.Interactions.Custom;assembly=Avalonia.Xaml.Interactions.Custom"
          x:Class="ControlCatalog.Pages.NotificationsPage">

		<Button>
			<i:Interaction.Behaviors>
				<ia:DataTriggerBehvior Binding="{Binding Message}" ComparisonCondition="Equal" Value="">
					<Setter Property="UseLayoutRounding" Value="True"/>
				</ia:DataTriggerBehvior>
			</i:Interaction.Behaviors>
			<TextBlock Text="Test" Foreground="Aqua"/>
		</Button>

</UserControl>

`
From csproj:

`
  <ItemGroup>
	  <PackageReference Include="Avalonia" Version="0.9.0" />
	  <PackageReference Include="Avalonia.ReactiveUI" Version="0.9.0" />
	  <PackageReference Include="Avalonia.Controls.DataGrid" Version="0.9.0" />
	  <PackageReference Include="Avalonia.Xaml.Behaviors" Version="0.9.0" />
	  <PackageReference Include="Avalonia.Xaml.Interactions" Version="0.9.0" />
   </ItemGroup>
`

I've also tried to upgrade to v. 0.10.17, which gave me a gazillion other errors as well as the one above.

So does anyone have a sample that actually works, or else an idea of what's wrong?

On https://docs.avaloniaui.net/guides/developer-guides/comparison-of-avalonia-with-wpf-and-uwp#styling it is claimed that DataTriggers doesn't work with Avalonia, which would render it useless IMHO. I'll also need MultiTriggers at some point, but haven't seen those.

Edit: I started out with: https://www.nequalsonelifestyle.com/2019/05/26/avalonia-controls-samples/ and tried to add DataTriggerBehavior there to test it.

**Comment by timunie** [comment, marked as answer]
See https://github.com/wieslawsoltes/AvaloniaBehaviors for examples and demos

Also make sure that it is installed as nuget packages, details in the link above :-)

**Reply by elveejay1** [reply]
Got it, and the demo works. I still have to figure out why my code which looks exactly the same fails with the error above.

**Reply by timunie** [reply]
Tip: try to create a minimal sample on your own and try to reproduce the issue there. Often this is much easier to debug.

**Reply by elveejay1** [reply]
Right - I took a simple MVVM-Avalonia project, and added a button similar to the above, and get the same error
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/9240488/AvaloniaApplication1.zip)

xaml:

`
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="using:AvaloniaApplication1.ViewModels"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:i="clr-namespace:Avalonia.Xaml.Interactivity;assembly=Avalonia.Xaml.Interactivity"
        xmlns:ia="clr-namespace:Avalonia.Xaml.Interactions.Core;assembly=Avalonia.Xaml.Interactions"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AvaloniaApplication1.Views.MainWindow"
        Icon="/Assets/avalonia-logo.ico"
        Title="AvaloniaApplication1">

    <Design.DataContext>
        <vm:MainWindowViewModel/>
    </Design.DataContext>
	<Grid>

		<TextBlock Text="{Binding Greeting}" HorizontalAlignment="Center" VerticalAlignment="Center">
			<TextBlock.Style>

			</TextBlock.Style>
		</TextBlock>

		<Button Name="TestBtn">
			<i:Interaction.Behaviors>
				<ia:DataTriggerBehvior Binding="{Binding Greeting}" ComparisonCondition="Equal" Value="">
					
					<ia:ChangePropertyAction TargetObject="TestBtn"
								  PropertyName="UseLayoutRounding"
								  Value="True" />
				</ia:DataTriggerBehvior>
			</i:Interaction.Behaviors>
		</Button>
	</Grid>

</Window>
`

csproj:

`
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
    
    <TrimMode>copyused</TrimMode>
    <BuiltInComInteropSupport>true</BuiltInComInteropSupport>
  </PropertyGroup>
  <ItemGroup>
    <Folder Include="Models\" />
    <AvaloniaResource Include="Assets\**" />
    <None Remove=".gitignore" />
  </ItemGroup>
  <ItemGroup>
    
    <TrimmableAssembly Include="Avalonia.Themes.Fluent" />
    <TrimmableAssembly Include="Avalonia.Themes.Default" />
  </ItemGroup>
  <ItemGroup>
    <PackageReference Include="Avalonia" Version="0.10.14" />
    <PackageReference Include="Avalonia.Desktop" Version="0.10.14" />
    
    <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="0.10.14" />
    <PackageReference Include="Avalonia.ReactiveUI" Version="0.10.14" />
    <PackageReference Include="XamlNameReferenceGenerator" Version="1.3.4" />
	  <PackageReference Include="Avalonia.Xaml.Behaviors" Version="0.10.14" />
	  <PackageReference Include="Avalonia.Xaml.Interactions" Version="0.10.14" />
  </ItemGroup>
</Project>
`

**Reply by timunie** [reply]
Allright, after debugging I found you are missing an `a`. It's `DataTriggerBehavior`, not `DataTriggerBehvior`

**Reply by elveejay1** [reply]
Doh...thanks. Strange, I was pretty sure I took it from intellisense (when intellisense worked - seems to not always be the case). This was somewhat embarresing, but at least it runs now...and I have some samples of the other behaviors to play with.
